src/v/raft/vote_stm.cc

```diff
@@ -189,8 +189,65 @@ ss::future<election_success> vote_stm::do_vote() {
     co_return election_success(_success);
 }
 
+bool vote_stm::has_request_in_progress() const {
+    return std::any_of(
+      _replies.begin(),
+      _replies.end(),
+      [](const absl::flat_hash_map<vnode, vmeta>::value_type& p) {
+          return p.second.get_state() == vmeta::state::in_progress;
+      });
+}
+
+bool vote_stm::can_wait_for_all() const {
+    return clock_type::now()
+           < _requests_dispatched_ts + _ptr->_jit.base_duration();
+}
+
 ss::future<> vote_stm::process_replies() {
     return ss::repeat([this] {
+        const bool request_in_progress = has_request_in_progress();
+        if (
+          (!request_in_progress || can_wait_for_all())
+          && _ptr->_enable_longest_log_detection()) {
+            if (request_in_progress) {
+                vlog(
+                  _ctxlog.debug,
+                  "[pre-vote {}] trying to wait for vote replies from all of "
+                  "the nodes",
+                  _prevote);
+                return with_gate(
+                         _vote_bg,
+                         [this] {
+                             return _sem.wait(_ptr->_jit.base_duration(), 1);
```

Comment:
we can only wait for 

(_requests_dispatched_ts + _ptr->_jit.base_duration() - clock_type::now()) ?

Reply:
i think it depends, the `vote_request` timeout is larger than the election timeout, indeed i can adjust wait timeout here to be dependent from the dispatch timestamp